catalog/mollie/mollie_helper.php

```diff
@@ -398,4 +401,67 @@ public function log ($transaction_id, $status, $order_id)
 			tep_db_query("INSERT INTO " . self::DB_PAYMENTS_TABLE . " (payment_id, status, osc_order_id) VALUES('" . $transaction_id . "', '" . $status . "', '" . $order_id . "')");
 		}
 	}
+
+	/**
+	 * Updates the order status and sends emails if enabled
+	 *
+	 * @param int $order_id 		 Internal order id
+	 * @param string $payment_status Mollie payment status
+	 */
+	public function update_status($order_id, $payment_status)
+	{
+		$new_status_id = static::get_status_id($payment_status);
+
+		$order_updated = false;
+		$check_status_query = tep_db_query("select customers_name, customers_email_address, orders_status, date_purchased from " . TABLE_ORDERS . " where orders_id = '" . (int)$order_id . "'");
+		$check_status = tep_db_fetch_array($check_status_query);
+		$customer_notified = '0';
+
+		if ( $new_status_id && $check_status['orders_status'] != $new_status_id ) 
+		{
+			if ( $this->get_status_notify($payment_status) ) 
+			{
+				require_once dirname(__FILE__) . "/../admin/includes/languages/english/orders.php";
+				require_once dirname(__FILE__) . "/../admin/includes/filenames.php";
+				$internal_status = $this->get_internal_status_name_from_id($new_status_id);
+				$email = STORE_NAME . "\n" . EMAIL_SEPARATOR . "\n" . EMAIL_TEXT_ORDER_NUMBER . ' ' . $order_id . "\n" . EMAIL_TEXT_INVOICE_URL . ' ' . tep_href_link(FILENAME_CATALOG_ACCOUNT_HISTORY_INFO, 'order_id=' . $order_id, 'SSL') . "\n" . EMAIL_TEXT_DATE_ORDERED . ' ' . tep_date_long($check_status['date_purchased']) . "\n\n" . sprintf(EMAIL_TEXT_STATUS_UPDATE, $internal_status);
+				tep_mail($check_status['customers_name'], $check_status['customers_email_address'], EMAIL_TEXT_SUBJECT, $email, STORE_OWNER, STORE_OWNER_EMAIL_ADDRESS);
+				$customer_notified = '1';
+			}
+
+			// Only update order status when order_status is different then the saved order_status
+			tep_db_query("UPDATE orders SET orders_status = '" . $new_status_id . "', last_modified = now() WHERE orders_id = " . intval($order_id));
+			$this->add_order_history($order_id, $payment_status, $customer_notified);
+		}
+	}
+
+	/**
+	 * Retrieve notify setting attached to status
+	 *
+	 * @param $status_name
+	 * @return bool|true
+	 */
+	protected static function get_status_notify ($status_name)
+	{
+		if (isset(static::$STATUS_NOTIFY[$status_name]) && static::$STATUS_NOTIFY[$status_name])
+		{
+			return static::$STATUS_NOTIFY[$status_name] == "True";
+		}
+
+		// fallback, default: send notification.
+		return true;
```

Comment:
Robert,

 

Ik heb de wijziging opgenomen. Heb een pull request aangemaakt.

 

Ik heb het idee dat er telkens iets mis gaat met de ‘tabs’ doordat we verschillende editors gebruiken.. Heb je een idee om dit permanent op te lossen?

 

George.

 

From: Robbert [mailto:notifications@github.com] 
Sent: woensdag 15 april 2015 12:05
To: mollie/osCommerce
Cc: George van Engers
Subject: Re: [osCommerce] Added confirmation e-mail (#1)

 

In catalog/mollie/mollie_helper.php <https://github.com/mollie/osCommerce/pull/1#discussion_r28407441> :

> +
> +     /**
> +      * Retrieve notify setting attached to status
> +      *
> +      * @param $status_name
> +      * @return bool|true
> +      */
> +     protected static function get_status_notify ($status_name)
> +     {
> +            if (isset(static::$STATUS_NOTIFY[$status_name]) && static::$STATUS_NOTIFY[$status_name])
> +            {
> +                    return static::$STATUS_NOTIFY[$status_name] == "True";
> +            }
> +
> +            // fallback, default: send notification.
> +            return true;

Zou je hier nog een check kunnen plaatsen, zodat standaard alleen een e-mail wordt verzonden bij status 'paid'?

—
Reply to this email directly or view it on GitHub <https://github.com/mollie/osCommerce/pull/1/files#r28407441> .  <https://github.com/notifications/beacon/ADrkE86NcfzhVp_V-HyWNpBNayQ8vuJUks5n_i9SgaJpZM4D_e1C.gif>